bigframes/core/compile/compiled.py

```diff
@@ -1062,18 +1062,24 @@ def _bake_ordering(self) -> OrderedIR:
                 )
                 new_baked_cols.append(baked_column)
                 new_expr = OrderingExpression(
-                    ex.free_var(baked_column.name), expr.direction, expr.na_last
+                    ex.free_var(baked_column.get_name()), expr.direction, expr.na_last
                 )
                 new_exprs.append(new_expr)
-            else:
+            elif isinstance(expr.scalar_expression, ex.UnboundVariableExpression):
                 new_exprs.append(expr)
+                new_baked_cols.append(self._ibis_bindings[expr.scalar_expression.id])
 
-        ordering = self._ordering.with_ordering_columns(new_exprs)
+        new_ordering = ExpressionOrdering(
+            tuple(new_exprs),
+            self._ordering.integer_encoding,
+            self._ordering.string_encoding,
+            self._ordering.total_ordering_columns,
+        )
         return OrderedIR(
             self._table,
             columns=self.columns,
-            hidden_ordering_columns=[*self._hidden_ordering_columns, *new_baked_cols],
```

Comment:
Is it intension to omit the existing `_hidden_ordering_columns` here?

Reply:
yes, we are rewriting the hidden columns here, so we are recreating the set of hidden columns and discarding the old set.